saleor/graphql/product/types/products.py

```diff
@@ -1094,13 +1132,63 @@ def resolve_image_by_id(root: ChannelContext[models.Product], _info, *, id):
         return root.node.media.filter(pk=pk).first()
 
     @staticmethod
-    def resolve_media(root: ChannelContext[models.Product], info):
-        return MediaByProductIdLoader(info.context).load(root.node.id)
+    def resolve_media(root: ChannelContext[models.Product], info, sort_by=None):
+        if sort_by is None:
+            sort_by = {
+                "field": ["sort_order"],
+                "direction": "",
+            }
+
+        def sort_media(media) -> list[ProductMedia]:
+            reversed = sort_by["direction"] == "-"
+            media_sorted = sorted(
+                media,
+                key=lambda x: tuple(getattr(x, field) for field in sort_by["field"]),
+                reverse=reversed,
+            )
+            return media_sorted
+
+        return MediaByProductIdLoader(info.context).load(root.node.id).then(sort_media)
 
     @staticmethod
     def resolve_images(root: ChannelContext[models.Product], info):
         return ImagesByProductIdLoader(info.context).load(root.node.id)
 
+    @staticmethod
+    def resolve_variant(root: ChannelContext[models.Product], info, id=None, sku=None):
+        if id is not None and sku is not None:
+            return None
```

Comment:
IMO we could use `validate_one_of_args_is_in_query` from [saleor/saleor/graphql/core/validators/__init__.py](https://github.com/mirumee/saleor/blob/a5c3dd97b5c81bb8121617ed019971fb96ab8172/saleor/graphql/core/validators/__init__.py#L21)
```suggestion
        validate_one_of_args_is_in_query("id", id, "slug", slug)
```